compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/bufferize_to_allocation_pack.mlir

```diff
@@ -0,0 +1,98 @@
+// RUN: iree-opt --pass-pipeline='builtin.module(func.func(iree-amdaie-bufferize-to-allocation{memory-space=1}))' --split-input-file %s | FileCheck %s --check-prefix=CHECK-1
+// RUN: iree-opt --pass-pipeline='builtin.module(func.func(iree-amdaie-bufferize-to-allocation{memory-space=2}))' --split-input-file %s | FileCheck %s --check-prefix=CHECK-2
+
```

Comment:
@kumardeepakamd this is the pass which we are pulling out and adding it separately since both pad and pack based approach use it.

[Here](https://github.com/nod-ai/iree-amd-aie/blob/main/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/pad_and_bufferize.mlir#L89-L112) is the current way this takes place in top-of-main for matmul pad based and the e2e for matmul pad based flow works on top-of-main.